### ISSUE TITLE
[Feature Request]: WriteToParquet schema as side input

### ISSUE THREAD

**Issue by EilidhMayne** [apache/beam#27164]
### What would you like to happen?

We have a need to pass a dynamic schema to the WriteToParquet PTransform but it will not accept a side input in the schema arg. Would this be possible to implement? My thought is that a similar approach to the WriteToBigQuery PTransform could be used.

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by rohan472000** [comment]
Yes same here ...I also want to pass a dynamic bucket location other than a static one.
